src/perception/test/standard_deviation_validator_test.cpp

```diff
@@ -0,0 +1,108 @@
+#include "clustering/dbscan.hpp"
+#include "cone_validator/deviation_validator.hpp"
+
+#include <gtest/gtest.h>
+
+
+/**
+ * @brief Fixture for testing the DeviationValidator class.
+ */
+class StandardDeviationTest : public ::testing::Test {
+ public:
+   /**
+   * @brief Set up the test fixtures.
+   */
+  void SetUp() override {
+    _point_cloud_.reset(new pcl::PointCloud<pcl::PointXYZI>);
+  }
+  pcl::PointCloud<pcl::PointXYZI>::Ptr _point_cloud_;
+  Plane _plane_;
+};
+
+/**
+ * @brief Test case to validate the an empty cluster
+ */
+TEST_F(StandardDeviationTest, NullPointCloud) {
+  auto deviation_validator = DeviationValidator(-1, 100, -1, 100);
+  auto cluster = Cluster(_point_cloud_);
+
+  ASSERT_FALSE(deviation_validator.coneValidator(&cluster, _plane_));
+}
+
+/**
+ * @brief Test case to validate the a cluster with the same Z
```

Comment:
What does the same Z mean?

Reply:
Same Z, means a standard deviation on z axis of 0. I will clarify that